src/components/BlogList.jsx

```diff
@@ -1,22 +1,46 @@
 import BlogPost from "./BlogPost";
 import Pagination from "./Pagination";
-import React from "react";
+import React, {useState} from "react";
 import blogs from "../data/blogs.json";
 
 const PAGE_SIZES = [15, 25, 50, 100];
 
 function BlogList() {
-  const currentPaginationData = blogs.posts.slice(0, 15);
 
-  const updateRowsPerPage = () => {};
-  const updatePage = () => {};
+  // state that stores current page index
+  const [currentPage, setCurrentPage] = useState(1);
+
+  
+
+  // state that stores number of blogs to show per page
+  const [selectedPageSize, setSelectedPageSize] = useState(15);
+  const [sliceAmount, setSliceAmount] = useState(15);
+  const [sliceIndex, setSliceIndex] = useState(0);
+  
+  // updates row per page based on user selection
+  const updateRowsPerPage = (userPageSize) => {
+    console.log(userPageSize);
+    const setUserPage = parseInt(userPageSize);
+    setSelectedPageSize(setUserPage)
+  };
+  // updates current page when user clicks onNext or onPrevious
+  const updatePage = (currentPage) => {
+    setCurrentPage(currentPage);
+    setSliceAmount(selectedPageSize*currentPage);
+    setSliceIndex((selectedPageSize*currentPage) - selectedPageSize);
+    console.log('slice amount', sliceAmount);
+  };
+
+  const currentPaginationData = blogs.posts.slice(((selectedPageSize*currentPage) - selectedPageSize) , selectedPageSize*currentPage);
```

Comment:
The "blogs.posts" array is being unnecessarily sliced on every render which is wasteful. This can be avoided by placing the sliced array in a state or memoizing it.